wimse/server/server.go

```diff
@@ -91,29 +95,29 @@ func (s *Server) getHttp() *http.Server {
 		pubInterface, err := x509.ParsePKIXPublicKey(keyBytes)
 		if err != nil {
 			log.Printf("failed to parse public key: %v", err)
-			http.Error(w, "Unauthorized", http.StatusUnauthorized)
+			wimseError(w)
 			return
 		}
 		clientEcdsa := pubInterface.(*ecdsa.PublicKey)
 
 		verifier, err := httpsign.NewP256Verifier(*clientEcdsa, httpsign.NewVerifyConfig().SetKeyID("wimse"), httpsign.Headers("@request-target", "Workload-Identity-Token"))
 		if err != nil {
 			log.Printf("Unable to create verifier: %v\n", err)
-			http.Error(w, "Unauthorized", http.StatusUnauthorized)
+			wimseError(w)
 			return
 		}
 
 		err = httpsign.VerifyRequest("wimse", *verifier, r)
 		if err != nil {
 			log.Printf("Invalid signature: %v\n", err)
-			http.Error(w, "Unauthorized", http.StatusUnauthorized)
+			wimseError(w)
 			return
 		}
 
 		svid, err := s.GetWITSVID()
 		if err != nil {
 			log.Printf("Unable to get WIT SVID: %v\n", err)
-			http.Error(w, "Unauthorized", http.StatusUnauthorized)
+			wimseError(w)
```

Comment:
Arguably from here down could be more internal server errors given we have processed the request and are building the response?